### PR TITLE
Add github pipeline to deploy frontend on S3 bucket

### PR DESCRIPTION
**Pull request by Utwo**
Add github pipeline to deploy frontend on S3 bucket.
Remove old frontend deployment strategy.

### CODE REVIEW

File: frontend/src/config/globals.jsx [hunk 1]
```diff
@@ -1,26 +1,5 @@
...
-    process.env.NODE_ENV === 'development',
-  isProd:
-    windowEnvironment.ENV === 'production' ||
-    process.env.NODE_ENV === 'production',
-
-  specifyUnknownData:
-    (windowEnvironment.SPECIFY_UNKNOWN_DATA ||
-      process.env.REACT_APP_SPECIFY_UNKNOWN_DATA ||
-      'false') === 'true',
+  api: process.env.REACT_APP_API_URL,
```
Comment (CristiHabliuc) [inline frontend/src/config/globals.jsx:2]: how will we pass the API url? at build time?

Comment (Utwo) [reply frontend/src/config/globals.jsx:2]: Yes, take a look here https://github.com/code4romania/date-la-zi/pull/306/files#diff-46f99ccdfc1ed31dc64ca065888b9236R45

Comment (CristiHabliuc) [reply frontend/src/config/globals.jsx:2]: Okay so we'll hard code them into the script

File: .github/workflows/deploy-frontend.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+      - name: Lint
+        working-directory: ${{env.working-directory}}
+        run: yarn lint
+
+      - name: Build
+        working-directory: ${{env.working-directory}}
+        run: yarn build
+        env:
+          REACT_APP_API_URL: "https://code4rocoviz19api-demo.azurewebsites.net/api/v2"
```
Comment (madpipeline) [inline .github/workflows/deploy-frontend.yml:45]: Why are we using a non commited backend in staging?

File: frontend/.env.example [hunk 3]
```diff
@@ -0,0 +1 @@
+REACT_APP_API_URL=https://code4rocoviz19api-demo.azurewebsites.net/api/v2
```
Comment (madpipeline) [inline frontend/.env.example:1]: What is the purpose of this file?

Comment (Utwo) [reply frontend/.env.example:1]: This is for development on frontend. Every user should copy .env.example in .env and change the file to the desired URL. Maybe I should update the set up part of the readme

File: .github/workflows/deploy-frontend.yml (hunk 2 above)
Comment (Utwo) [reply .github/workflows/deploy-frontend.yml:45]: This was the API that I used for development so I thought it was ok to use this on staging. I don't know so much about how backend prod and stage is structured.

File: frontend/.env.example (hunk 3 above)
Comment (madpipeline) [reply frontend/.env.example:1]: :+1:

File: frontend/src/config/globals.jsx (hunk 1 above)
Comment (madpipeline) [reply frontend/src/config/globals.jsx:2]: FE not using the API anymore

**Comment by madpipeline** [comment]
I've done the following:

* streamlined the GHA pipeline to have a single flow
* Added CloudFront to serve from S3
* removed Frontend Fargate service

I haven't been able to configure CloudFront to use custom domains, to use our domains, so this PR is not done yet, but it's close.

File: aws_deploy/cache.tf [hunk 4]
```diff
@@ -0,0 +1,53 @@
+resource "aws_cloudfront_distribution" "main" {
+  enabled         = true
+  is_ipv6_enabled = true
+  price_class     = "PriceClass_100"
+  aliases         = terraform.workspace == "production" ? [local.domain_root, local.domain_frontend] : [local.domain_frontend]
```
Comment (madpipeline) [inline aws_deploy/cache.tf:5]: This is not working. We need a slightly different approach: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/CNAMEs.html#alternate-domain-names-requirements

**Comment by madpipeline** [comment]
Making progress, but I'm still having some issues: https://discuss.hashicorp.com/t/provider-alias-errors-with-provider-configuration-not-present/8271

**Pull request merged**